src/MarDyn.cpp

```diff
@@ -132,8 +132,20 @@ int run_unit_tests(const Values &options, const std::vector<std::string> &args)
  */
 int main(int argc, char** argv) {
 #ifdef ENABLE_MPI
-	MPI_Init(&argc, &argv);
-#endif
+#if defined(_OPENMP)
+	const int thread_support_level_requested = MPI_THREAD_FUNNELED; /* So far only the main thread makes MPI calls */
+#else
+	const int thread_support_level_requested = MPI_THREAD_SINGLE; /* No thread support required */
+#endif /* _OPENMP */
```

Comment:
```suggestion

	const int thread_support_level_requested = 
#if defined(_OPENMP)
		MPI_THREAD_FUNNELED; /* So far only the main thread makes MPI calls */
#else
		MPI_THREAD_SINGLE; /* No thread support required */
#endif /* _OPENMP */
```

Arguably a question of style, but I prefer to `ifdef` only what is really different.

Reply:
I never thought of this way to write it - but I must say I find it much harder to read as the variable name is not in the same line as the value, which makes my eyes jump over more lines of code than I want and my brain has to keep track of the ifdef in the meanwhile ...